File: backend/app/schemas/email_generator.py
from pydantic import BaseModel
from typing import Literal

class EmailGenerateRequest(BaseModel):
    resume_id: int
    job_id: int
    email_type: Literal["follow-up", "thank-you", "withdrawal"]
    tone: Literal["formal", "casual", "enthusiastic"]
    interviewer_name: str | None = None   # optional, used in thank-you
    days_since_applied: int | None = None # optional, used in follow-up

class EmailGenerateResponse(BaseModel):
    subject: str
    body: str
    email_type: str
    tone: str
    company: str
    role: str

File: frontend/src/api/emailGenerator.ts
import axios from "axios";

const API_URL = import.meta.env.VITE_API_URL || "http://localhost:8000";

// Make sure ALL these have the export keyword
export type EmailType = "follow-up" | "thank-you" | "withdrawal";
export type EmailTone = "formal" | "casual" | "enthusiastic";

export interface EmailGenerateRequest {
  resume_id: number;
  job_id: number;
  email_type: EmailType;
  tone: EmailTone;
  interviewer_name?: string;
  days_since_applied?: number;
}

export interface EmailGenerateResponse {  // ← make sure 'export' is here
  subject: string;
  body: string;
  email_type: string;
  tone: string;
  company: string;
  role: string;
}

export const generateEmail = async (
  payload: EmailGenerateRequest
): Promise<EmailGenerateResponse> => {
  const token = localStorage.getItem("token");
  const response = await axios.post(
    `${API_URL}/email/generate`,
    payload,
    { headers: { Authorization: `Bearer ${token}` } }
  );
  return response.data;
};